client/client-lib/src/api.rs

```diff
@@ -214,51 +186,75 @@ impl ApiError {
     }
 }
 
-#[cfg_attr(target_family = "wasm", async_trait(?Send))]
+#[cfg_attr(target_family = "wasm", async_trait(? Send))]
 #[cfg_attr(not(target_family = "wasm"), async_trait)]
 impl<C: JsonRpcClient + Send + Sync> FederationApi for WsFederationApi<C> {
     /// Fetch the outcome of an entire transaction
     async fn fetch_tx_outcome(&self, tx: TransactionId) -> Result<TransactionStatus> {
-        self.request("/fetch_transaction", tx).await
+        self.request("/fetch_transaction", tx, &mut Retry404::default())
+            .await
     }
 
-    /// Submit a transaction to all federtion members
+    /// Submit a transaction to all federation members
     async fn submit_transaction(&self, tx: Transaction) -> Result<TransactionId> {
         // TODO: check the id is correct
-        self.request("/transaction", tx).await
+        self.request("/transaction", tx, &mut CurrentConsensus::default())
+            .await
     }
 
-    async fn fetch_epoch_history(&self, epoch: u64) -> Result<EpochHistory> {
-        self.request("/fetch_epoch_history", epoch).await
+    async fn fetch_epoch_history(&self, epoch: u64, epoch_pk: PublicKey) -> Result<EpochHistory> {
+        self.request(
+            "/fetch_epoch_history",
+            epoch,
+            &mut ValidHistory::new(epoch_pk),
+        )
+        .await
     }
 
     async fn fetch_contract(&self, contract: ContractId) -> Result<ContractAccount> {
-        self.request("/ln/account", contract).await
+        self.request("/ln/account", contract, &mut Retry404::default())
+            .await
```

Comment:
Consistency requirements seems dependent on use case, did you check if retry is good enough for all current ones?

Reply:
Shouldn't consistency of `f+1` peers be sufficient in all cases?